frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/RequestManager.java

```diff
@@ -406,6 +406,16 @@ public PlatformSpecificParametersDescriptor getPlatParams() {
     return taskResources;
   }
 
+  public int getTotalGpuCount() {
+    int gpuCount = 0;
+    Map<String, TaskRoleDescriptor> cachedTaskRoles = taskRoles;
+    for (String taskRoleName : taskResources.keySet()) {
```

Comment:
>keySet [](start = 45, length = 6)

entrySet

Reply:
If so, just use valueSet?

---
In reply to: [174689838](https://github.com/Microsoft/pai/pull/318#discussion_r174689838) [](ancestors = 174689838)